### PR TITLE
Fix 9123 - [UWP] Use style to show BackgroundColor on Button / ImageButton when hovering

### LINKED ISSUE #9123: [Bug] UWP Button BackgroundColor turns gray when hovering
### Description
UWP Button BackgroundColor turns gray when hovering ono it while targetting builds: 17763 and 18362, problem doesn't appear in creators update (16299)

### Steps to Reproduce

1. Create a XF button with background color
2. target UWP 17763 or 18362
3. hover on the button

### Expected Behavior
Button preserves its background color (similar to what we get when targetting version 16299)

### Actual Behavior
Button turns to gray (or whatever default theme)

### Basic Information
  - UWP: SDK versions: 17763 and 18362

### PR DESCRIPTION
**Pull request by WayaFlyfeather**
### Description of Change ###

By applying a style (a copy of the default Button style in UWP version 16399) instead of relying on the default Button style, this PR ensures that setting BackgroundColor for a Button or an ImageButton is respected when the mouse pointer is hovered above the Button / ImageButton. Without this the Button or ImageButton BackgroundColor turns grey when hovering the mouse pointer on UWP TargetPlatformVersion 17763 or above, due to changes in the default Button style's ControlTemplate.
 
### Issues Resolved ### 

- fixes #9123 

### API Changes ###
 
 None

### Platforms Affected ### 

- UWP

### Behavioral/Visual Changes ###
On UWP 17763 and above, BackgroundColor for Button and ImageButton wil be respected when hovering mouse pointer, instead of turning grey.

### Before/After Screenshots ### 
Screenshots from UWP TargetPlatformVersion 18362:

Before fix, no hover:
fix-9123-before-no-hover

Before fix, hover over Button:
fix-9123-before-hover-button

Before fix, hover over ImageButton:
fix-9123-before-hover-image-button

After fix, no hover:
fix-9123-after-no-hover

After fix, hover over Button:
fix-9123-after-hover-button

After fix, hover over ImageButton:
fix-9123-after-hover-image-button

### Testing Procedure ###
Be sure to compile and run the ControlGallery app with UWP TargetPlatformVersion 17763 or above. Go to Test Case 9123. Hover the mouse pointer over the Button and the ImageButton, and make sure the BackgroundColor remains yellow / yellow-green.

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by jsuarezruiz** [comment]
@WayaFlyfeather Could you rebase 4.6.0?. There are some conflicts, thanks.

File: Xamarin.Forms.Platform.UAP/ImageButtonRenderer.cs [hunk 1]
```diff
@@ -71,7 +71,11 @@ protected async override void OnElementChanged(ElementChangedEventArgs<ImageButt
 			{
 				if (Control == null)
 				{
-					_formsButton = new FormsButton();
+					_formsButton = new FormsButton
+					{
+						Style = Windows.UI.Xaml.Application.Current.Resources["FormsButtonStyle"] as Windows.UI.Xaml.Style
+					};
+
 					_formsButton.Padding = new WThickness(0);
```
Comment (nickrandolph) [inline Xamarin.Forms.Platform.UAP/ImageButtonRenderer.cs:79]: Shouldn't these properties (Padding, BorderThickness and Background) be set in the Style now?

File: Xamarin.Forms.Platform.UAP/FormsButtonStyle.xaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                            <VisualStateGroup x:Name="CommonStates">
+                                <VisualState x:Name="Normal">
+                                    <Storyboard>
+                                        <PointerUpThemeAnimation Storyboard.TargetName="RootGrid"/>
+                                    </Storyboard>
+                                </VisualState>
+                                <VisualState x:Name="PointerOver">
+                                    <Storyboard>
+                                        <ObjectAnimationUsingKeyFrames Storyboard.TargetName="RootGrid" Storyboard.TargetProperty="Background">
+                                            <DiscreteObjectKeyFrame KeyTime="0" Value="{ThemeResource ButtonBackgroundPointerOver}"/>
```
Comment (nickrandolph) [inline Xamarin.Forms.Platform.UAP/FormsButtonStyle.xaml:34]: Doesn't this style still have the same issue with pointerover behaviour changing the background to grey?

**Comment by jfversluis** [comment]
I think we did some fixes in this area already, thank you for all the time and effort on this one everyone! Let me know if there is still an issue here

**Pull request closed without merging**